qiskit_optimization/applications/sk_model.py

```diff
@@ -0,0 +1,110 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the Sherrington Kirkpatrick (SK) model."""
+
+
+from typing import Union, List
+
+import networkx as nx
+import numpy as np
+from docplex.mp.model import Model
+
+from .graph_optimization_application import OptimizationApplication
+from ..algorithms import OptimizationResult
+from ..problems.quadratic_program import QuadraticProgram
+from ..translators import from_docplex_mp
+
+
+class SKModel(OptimizationApplication):
+    r"""Optimization application of the "Sherrington Kirkpatrick (SK) model" [1].
+
+    The SK Hamiltonian over n spins is given as:
+        :math:`H(x)=-1/\sqrt{n} \sum_{i<j} w_{i,j}x_ix_j`, :math:`where x_i\in\{\pm 1\}`
+    and :math:`w_{i,j}\in\{\pm 1\}` are chosen independently and uniformly at random.
+    Notice that there are other variants e.g., with :math:`w_{i,j}` chosen from the normal distribution
+    with mean 0 and variance 1.
+
+    References:
+        [1]: Dmitry Panchenko. "The Sherrington-Kirkpatrick model: an overview",
+        https://arxiv.org/abs/1211.1094
+    """
+
+    def __init__(self, num_of_sites: int, rng: np.random.RandomState = np.random.RandomState(0)):
+        """
+        Constructor for the SK model.
+
+        Args:
+            num_of_sites: number of sites
+            rng: numpy pseudo-random number generator
+        """
+        self._rng = rng
+        self._num_of_sites = num_of_sites
+        self._graph = nx.complete_graph(self._num_of_sites)
+
+        self.new_instance()
+
+    def new_instance(self) -> None:
+        """Generate a new instance of the SK model."""
+        for _, _, edge_data in self._graph.edges(data=True):
+            edge_data["weight"] = self._rng.choice([-1, 1])
+
+    def to_quadratic_program(self) -> QuadraticProgram:
+        """Convert an SK model problem instance into a
+        :class:`~qiskit_optimization.problems.QuadraticProgram`.
+
+        Returns:
+            The :class:`~qiskit_optimization.problems.QuadraticProgram` created
+            from the SK problem instance.
+        """
+        mdl = Model(name="SK-model")
+        x = {
+            i: mdl.binary_var(name="x_{0}".format(i)) for i in range(self._graph.number_of_nodes())
+        }
```

Comment:
```suggestion
        x = mdl.binary_var_list(self._graph.number_of_nodes())
```

Reply:
I guess that this should be then refactored also for other problems in this module.